osc_preferences.h

```diff
@@ -0,0 +1,126 @@
+#ifndef __OSC_PREFERENCES_H__
+#define __OSC_PREFERENCES_H__
+
+#include <glib/glist.h>
+#include <glib/gprintf.h>
+
+typedef struct osc_preferences OscPreferences;
+typedef struct oscplot_preferences OscPlotPreferences;
+typedef struct iio_device_preferences IioDevPreferences;
+typedef struct iio_channel_preferences IioChnPreferences;
+
+struct oscplot_preferences {
+        unsigned int *sample_count;
+        GList *devices_pref_list;
+};
+
+struct iio_channel_preferences {
+        char *chn_name;
+        unsigned int *constraints;
+};
+
+struct iio_device_preferences {
+        char *dev_name;
+        double *fft_correction;
+        GList *channels_pref_list;
+};
+
+struct osc_preferences {
+        struct oscplot_preferences *plot_preferences;
+};
+
+
+/* Construct/Destruct methods */
+
+
+/* IioChnPreferences */
+
+static inline IioChnPreferences* iio_chn_preferences_new(const char *chn_name)
+{
+        IioChnPreferences *chn_pref = g_new(IioChnPreferences, 1);
+
+        chn_pref->chn_name = g_strdup(chn_name);
+        chn_pref->constraints = NULL;
+
+        return chn_pref;
+}
+
+static inline void iio_chn_preferences_free(IioChnPreferences *pref)
+{
+        if (pref->chn_name) {
+                g_free(pref->chn_name);
+                pref->chn_name = NULL;
+        }
+}
+
+/* IioDevPreferences */
+
+static inline IioDevPreferences* iio_dev_preferences_new(const char *dev_name)
+{
+        IioDevPreferences *dev_pref = g_new(IioDevPreferences, 1);
+
+        dev_pref->dev_name = g_strdup(dev_name);
+        dev_pref->fft_correction = NULL;
+        dev_pref->channels_pref_list = NULL;
+
+        return dev_pref;
+}
+
+static inline void iio_dev_preferences_free(IioDevPreferences *pref)
+{
+        if (pref->dev_name) {
+                g_free(pref->dev_name);
+                pref->dev_name = NULL;
+        }
+
+        if (pref->fft_correction) {
+                g_free(pref->fft_correction);
+                pref->fft_correction = NULL;
+        }
+
+        g_list_free_full(pref->channels_pref_list,
+                (GDestroyNotify)iio_chn_preferences_free);
+}
+
+/* OscPlotPreferences */
+
+static inline OscPlotPreferences* osc_plot_preferences_new()
+{
+        OscPlotPreferences *plot_pref = g_new(OscPlotPreferences, 1);
+
+        plot_pref->sample_count = NULL;
+        plot_pref->devices_pref_list = NULL;
```

Comment:
just curios; does `g_new()` does [or doesn't] guarantee that the memory is zeroed-out ?

if it doesn't guarantee, one way to do this consistently is to use a `memset(plot_pref, 0)` type of construct;
this will be future-proof when you want to add new field, but forget to initialize the NULL here;

Reply:
I can use g_new0() which does just that. Thanks!